identity/src/main/java/com/stripe/android/identity/navigation/ErrorDestination.kt

```diff
@@ -0,0 +1,77 @@
+package com.stripe.android.identity.navigation
+
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
+
+internal class ErrorDestination(
+    errorTitle: String,
+    errorContent: String,
+    backButtonText: String,
+    backButtonDestination: String = UNEXPECTED_ROUTE,
+    shouldFail: Boolean = false
+) : IdentityTopLevelDestination() {
+
+    override val destinationRoute = ROUTE
+    override val routeWithArgs = destinationRoute.withParams(
+        ARG_ERROR_TITLE to errorTitle,
+        ARG_ERROR_CONTENT to errorContent,
+        ARG_GO_BACK_BUTTON_DESTINATION to backButtonDestination.toRouteBase(),
+        ARG_GO_BACK_BUTTON_TEXT to backButtonText,
+        ARG_SHOULD_FAIL to shouldFail
+    )
+
+    internal companion object {
+        const val ERROR = "Error"
+        const val ARG_ERROR_TITLE = "errorTitle"
+        const val ARG_ERROR_CONTENT = "errorContent"
+
+        // if set, shows go_back button, clicking it would navigate to the destination.
+        const val ARG_GO_BACK_BUTTON_TEXT = "goBackButtonText"
+        const val ARG_GO_BACK_BUTTON_DESTINATION = "goBackButtonDestination"
+
+        // if set to true, clicking bottom button and pressBack would end flow with Failed
+        const val ARG_SHOULD_FAIL = "shouldFail"
+
+        // Indicates the server returns a requirementError that doesn't match with current route.
+        //  E.g ConsentScreen->DocSelectScreen could only have BIOMETRICCONSENT error but not IDDOCUMENTFRONT error.
+        // If this happens, set the back button destination to [DEFAULT_BACK_BUTTON_DESTINATION]
+        const val UNEXPECTED_ROUTE = "UnexpectedRoute"
+
+        fun errorTitle(backStackEntry: NavBackStackEntry) =
```

Comment:
the parameters passed to a destination is set with `destinationRoute.withParams` method.
`destinationRoute` is something like `Error?arg1={arg1}&arg2={arg2}`
`destinationRoute.withParams` set the params with actual values `Error?arg1=value1&arg2=value2`.

This would in turn created a `NavBackStackEntry` with a bundle with `"arg1"="value1", "arg2"="value2"` , then we created these static function to fetch corresponding entries. This is a slightly generalized version of FinancialConnection's `ManualEntrySuccess.microdeposits`